numpy/core/getlimits.py

```diff
@@ -9,19 +9,198 @@
 from . import numeric
 from . import numerictypes as ntypes
 from .numeric import array
+from .umath import nextafter, log2, log10, isnan, exp2
+
 
 def _frz(a):
     """fix rank-0 --> rank-1"""
     if a.ndim == 0:
         a.shape = (1,)
     return a
 
+
 _convert_to_float = {
     ntypes.csingle: ntypes.single,
     ntypes.complex_: ntypes.float_,
     ntypes.clongfloat: ntypes.longfloat
     }
 
+
+def _get_next(ftype, start, above):
+    start = ftype(start)
+    above = ftype(above)
+    with numeric.errstate(all='ignore'):
+        return nextafter(start, above)
+
+
+def _get_eps(ftype, direction=1):
+    start = ftype(1)
+    direction = ftype(direction)
+    above = start + direction
+    return (nextafter(start, above) - start) * direction
```

Comment:
For negative direction this is going to depend on rounding of the difference. If you actually used that option it would bother me more ;)